### DISCUSSION TITLE
Advice Post-Bitmap for JPG Converted Metadata

### DISCUSSION THREAD

**Discussion by matthew-t-smith** [dlemstra/Magick.NET#1576] [Help]
I initially created an issue ticket here thinking that MagickImage might be missing an image's density (DPI) information: https://github.com/dlemstra/Magick.NET/issues/1575

Where I'm at now is that I believe the photo was previously managed as a TIFF and was compressed to a JPG:
![image](https://github.com/dlemstra/Magick.NET/assets/15874467/f8a25429-6487-440c-bfc2-328ccc0c7ce0)
The screenshot here shows via Adobe Bridge the raw image's metadata carries TIFF resolution information still. Simply converting the file back over to TIFF (as proof of concept only, realizing that would be bad practice) didn't yield any results (i.e. `image.Density` still equaled `{0, 0}`).

My next try was with .NET's Bitmap class which managed to read the data:
```
var bitmap = new Bitmap(filePath);
Console.WriteLine($"X: {bitmap.HorizontalResolution}, Y: {bitmap.VerticalResolution}");
// X: 72, Y: 72
```

Unfortunately here, Bitmap is now Windows only since .NET 6.

So, I'm needing some advice as to how I might be able to get at this density metadata information on the `.jpg` version of the file without re-converting or using another 3rd party library.

**Comment by dlemstra** [comment]
You can do something like this:
```#
var image = new MagickImage();
image.Ping(path);
// do something with image.Density;
```

But I have the feeling that I am misunderstanding your question.

**Comment by matthew-t-smith** [comment]
It seems that the compressed JPG version of this file (I've been told was maybe at least a `.psd` if not `.tiff` ) has carried some metadata from its earlier days (seen in the screenshot from Adobe Bridge's raw metadata XMP viewer). However, I can't seem to get at it from ImageMagick. I'm not familiar enough with how that would be buried in the file, but maybe TIFF info isn't taken into consideration - reasonably so - for a `.jpg`...

Regardless, good to know about `.Ping()` but that didn't get it either.
